sqlglot/expressions.py

```diff
@@ -234,6 +234,9 @@ def type(self) -> t.Optional[DataType]:
 
     @type.setter
     def type(self, dtype: t.Optional[DataType | DataType.Type | str]) -> None:
+        self.set_type(dtype)
+
+    def set_type(self, dtype: t.Optional[DataType | DataType.Type | str]) -> None:
```

Comment:
Why was this needed?

Reply:
mypy doesn't like
```python
expression.set = DataType.Type.Foo
```
Seems to be using the type of the getter, not the setter?

Reply:
just do type ignore then